lib/composer.js

```diff
@@ -20,45 +22,145 @@ export default class Composer extends Disposable {
+    this.initializeBuildStateFromProperties(state, atom.config.get('latex'))
+  }
+
+  initializeBuildStateFromProperties (state, properties) {
+    if (!properties) return
+
+    if (properties.cleanPatterns) {
+      state.setCleanPatterns(properties.cleanPatterns)
+    }
 
-      atom.notifications.addWarning(message, {
-        dismissable: true, description, buttons: [{ text: title, onDidClick: openUrl }]
-      })
+    if ('enableSynctex' in properties) {
+      state.setEnableSynctex(properties.enableSynctex)
     }
 
-    state.jobnames = state.builder.getJobNamesFromMagic(state.rootFilePath)
+    if ('enableShellEscape' in properties) {
+      state.setEnableShellEscape(properties.enableShellEscape)
+    }
+
+    if ('enableExtendedBuildMode' in properties) {
+      state.setEnableExtendedBuildMode(properties.enableExtendedBuildMode)
+    }
+
+    if (properties.jobNames) {
+      state.setJobNames(properties.jobNames)
+    } else if (properties.jobnames) {
+      // jobnames is for compatibility with magic comments
+      state.setJobNames(properties.jobnames)
+    } else if (properties.jobname) {
+      // jobname is for compatibility with Sublime
+      state.setJobNames([properties.jobname])
+    }
+
+    if (properties.customEngine) {
+      state.setEngine(properties.customEngine)
+    } else if (properties.engine) {
+      state.setEngine(properties.engine)
+    } else if (properties.program) {
+      // program is for compatibility with magic comments
+      state.setEngine(properties.program)
+    }
+
+    if ('moveResultToSourceDirectory' in properties) {
+      state.setMoveResultToSourceDirectory(properties.moveResultToSourceDirectory)
+    }
+
+    if (properties.outputFormat) {
+      state.setOutputFormat(properties.outputFormat)
+    } else if (properties.format) {
+      // format is for compatibility with magic comments
+      state.setOutputFormat(properties.format)
+    }
+
+    if ('outputDirectory' in properties) {
+      state.setOutputDirectory(properties.outputDirectory)
+    } else if ('output_directory' in properties) {
+      // output_directory is for compatibility with Sublime
+      state.setOutputDirectory(properties.output_directory)
+    }
 
-    return state
+    if (properties.producer) {
+      state.setProducer(properties.producer)
+    }
+  }
+
+  initializeBuildStateFromMagic (state) {
+    let magic = this.getMagic(state)
+
+    if (magic.root) {
+      state.setFilePath(path.resolve(state.getProjectPath(), magic.root))
+      magic = this.getMagic(state)
+    }
+
+    this.initializeBuildStateFromProperties(state, magic)
+  }
+
+  getMagic (state) {
+    return new MagicParser(state.getFilePath()).parse()
+  }
+
+  initializeBuildStateFromProjectFile (state) {
+    try {
+      const { dir, name } = path.parse(state.getFilePath())
+      const filePattern = path.format({ dir, name, ext: '.@(yaml|yml)' })
+      const filePath = _.first(glob.sync(filePattern))
```

Comment:
Since the official recommendation is to use `.yaml`, let's pick that? I don't really care either way, but there is no reason we need to support both extensions.